neo4j-cypher-dsl/src/main/java/org/neo4j/cypherdsl/core/OngoingListBasedPredicateFunction.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2019-2024 "Neo4j,"
+ * Neo4j Sweden AB [https://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.cypherdsl.core;
+
+import org.jetbrains.annotations.NotNull;
+import org.neo4j.cypherdsl.core.annotations.CheckReturnValue;
+
+/**
+ * Allows to define the source of the list predicate.
+ *
+ * @author Michael J. Simons
+ * @since 2024.0.0
+ */
+public interface OngoingListBasedPredicateFunction {
+
+	/**
+	 * @param list A list
```

Comment:
```suggestion
	 * @param list A list expression
```
I see that this was just extracted but let's give the param some more meaning :)